### PR TITLE
linkemulation

### PR DESCRIPTION
**Pull request by brentondwalker**
We are working on link emulation tools with MoonGen.  The main change so far in libmoon is adding more flexible ring buffers to hold frames and emulate latency.

### CODE REVIEW

File: lua/pipe.lua [hunk 1]
```diff
@@ -29,18 +29,170 @@ ffi.cdef [[
 	
 	// DPDK SPSC ring
 	struct rte_ring { };
+	//struct rte_mbuf { };
```
Comment (emmericp) [inline lua/pipe.lua]: this seems odd; did we define this twice somewhere accidentally?

**Comment by emmericp** [comment]
this looks great, thanks! :)

can you clean up the commit history a little bit? Just squashing it together to one commit should be fine here I guess

File: lua/pipe.lua (hunk 1 above)
Comment (brentondwalker) [reply lua/pipe.lua]: Not sure where this came from.  Removing the commented line.

**Comment by brentondwalker** [comment]
The commit and merge history of my fork was making rebase too complicated.  I combined the changes into one commit manually.

**Comment by emmericp** [comment]
thanks!

**Pull request merged**